### ISSUE TITLE
i18nTextCollector produces corrupt output / namespaces when running under PHP8.0

### ISSUE THREAD

**Issue by bimthebam** [silverstripe/silverstripe-framework#10227]
## Affected Version

silverstripe/framework 4.10

## Description

#9665 has introduced PHP8 support - especially for the new PHP parser tokens (`T_NAME_FULLY_QUALIFIED ` and `T_NAME_QUALIFIED`) - but is broken when collecting text from classes which are having `use` statements and even name-spaced method calls.

## Steps to Reproduce

1. Create a simple class which calls the `_t` function, has `use` statements and makes some name-spaced method calls. e.g.

```php
<?php

namespace App\Model;

use SilverStripe\ORM\DataObject;

/**
 * Class TestModelWithTFunctionCalls
 * @package App\Model
 */
class TestModelWithTFunctionCalls extends DataObject
{
    private static $table_name = 'TestModelWithTFunctionCalls';

    public function fieldLabels($includeRelations = true)
    {
        $labels = parent::fieldLabels(true);

        $currentLocale = \SilverStripe\i18n\i18n::get_locale();

        $labels['Test'] = _t(__CLASS__ . '.TEST', 'Test');

        try {

        } catch (\Exception $e) {

        }

        return $labels;
    }
}
```

2. Run the i18nTextCollector task.

As the class resists within the app directory:

```shell
vendor/bin/sake dev/tasks/i18nTextCollectorTask module=app locale=de
```

3. View the generated language file (`app/lang/de.yml`)

### Actual result

Mind the generated namespace (`App\Model\SilverStripe\ORM\DataObject\TestModelWithTFunctionCalls\\SilverStripe\i18n\i18n\get_locale`)

```yml
de:
  App\Model\SilverStripe\ORM\DataObject\TestModelWithTFunctionCalls\\SilverStripe\i18n\i18n\get_locale:
    TEST: Test
  App\Model\TestModelWithTFunctionCalls:
    PLURALNAME: 'Test Model With T Function Callss'
    PLURALS:
      one: 'A Test Model With T Function Calls'
      other: '{count} Test Model With T Function Callss'
    SINGULARNAME: 'Test Model With T Function Calls'
```

### Expected result

```yml
de:
  App\Model\TestModelWithTFunctionCalls:
    PLURALNAME: 'Test Model With T Function Callss'
    PLURALS:
      one: 'A Test Model With T Function Calls'
      other: '{count} Test Model With T Function Callss'
    SINGULARNAME: 'Test Model With T Function Calls'
    TEST: Test
```

**Comment by GuySartorelli** [comment]
Linked PR is merged. Closing this issue.

**Comment by blueskies79** [comment]
Hello! I see that https://github.com/silverstripe/silverstripe-framework/pull/10228 was merged at some point. But I'm currently using 4.11.6 and I'm running into the exact same problem as OP. 
I also see that the file changes in this pull request are not present in 4.11.6: [src/i18n/TextCollection/i18nTextCollector.php](https://github.com/silverstripe/silverstripe-framework/blob/4.11.6/src/i18n/TextCollection/i18nTextCollector.php) (line 574 thru 580). Is there a reason this was removed? Because the original problem has now returned with php 8

**Comment by GuySartorelli** [comment]
@blueskies79 Thanks for pointing that out. It was merged into `4` but didn't make the window for the `4.11.0` release. It would be released with `4.12.0` (which might be released in December) but since it's a straight forward fix I've backported it and released [4.11.10](https://github.com/silverstripe/silverstripe-framework/releases/tag/4.11.10).

**Comment by blueskies79** [comment]
@GuySartorelli Awesome, thanks so much!
